### ISSUE TITLE
GitHub checks "Re-run" button doesn't work

### ISSUE THREAD

**Issue by jonathan-j-lee** [web-platform-tests/wpt#39458]
In GitHub, there's a "Re-run" button next to failed checks:

![image](https://user-images.githubusercontent.com/16431277/231002476-8b279abb-772b-4e5c-b0d2-29fbaaef66ae.png)

However, clicking this button will add a comment to the PR/commit saying the request failed with a 403 error:

![image](https://user-images.githubusercontent.com/16431277/231003227-6ad76a38-77e9-4d8b-9158-0f457ad50929.png)

The "official" but undocumented way to re-run failed checks is to close and reopen the PR, but this is nonobvious to many contributors (e.g., #39241, #39443). The button should either work or not exist.

cc @TalbotG

**Comment by TalbotG** [comment]
Thank you, Jonathan.

**Comment by jgraham** [comment]
AFAIK we can't make the button not exist.

The problem with rerunning tasks in taskcluster is that it's not really clear how it should interact with dependent tasks. If the dependent task has already run it should probably be rescheduled, but updated with the new information about the parent task. If it hasn't been run it should also be updated with the new task information, but just run once. I don't think any of this logic to update task subtrees is implemented, and instead there are two definitions of rerun: run the same task with a different task id (which then doesn't affect dependents at all), or rerun the same task with the same task id (which is then subject to the same scheduling as the original task, and still won't rerun dependents that already ran). In particular for our purposes neither option causes the sink task to rerun, and that's the one that's required to make the GH checks pass.

So basically, even if the button did something, it would require significant changes in the upstream CI provider to make it do the expected thing. So if removing the button was possible that would be the most obvious option, but I don't think it is possible.

**Comment by past** [comment]
There is an [unanswered question](https://github.com/orgs/community/discussions/28179) about this in the GitHub community discussions, so indeed it doesn't look possible today.
